ntbs-service/Middleware/AuditGetRequestMiddleware.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using ntbs_service.Services;
+
+namespace ntbs_service.Middleware
+{
+    public class AuditGetRequestMiddleWare
+    {
+        private readonly RequestDelegate _next;
+
+        public AuditGetRequestMiddleWare(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context, IAuditService auditService)
+        {
+            var request = context.Request;
+
+            if (request.Method == "GET" && request.Query.ContainsKey("id"))

```

Comment:
Maybe this could be made prettier, but this should ensure that we log all get requests for viewing a particular entity (as the id field is present) - search pages etc will be ignored by this as required